examples/models/data_tables_server.py

```diff
@@ -23,10 +27,14 @@ def __init__(self):
         self.class_filter = None
 
         self.source = ColumnDataSource()
+        # A bit hacky, but now we have an index column that will carry through our slicing
+        self.df = DataFrame.from_dict(ColumnDataSource.from_df(mpg))
+        self.df.set_index('index', inplace=True)
         self.update_data()
-
-        self.document.add_root((self.create()))
-        self.session = push_session(self.document)
+        self.source.on_change('data', self.update_from_table_change)
+        # XXX: I haven't had much luck running the server in this style
+        # self.document.add_root((self.create()))
+        # self.session = push_session(self.document)
```

Comment:
Sorry for the slow response! My notifications got buried in my email somehow... Anyways, I see what you're saying with using the Client APIs. I have not tried to use them as you describe. For the example I made I was attempting to show the `.on_change` callback capability that I believe is only available with `bokeh serve`. 

From what you describe, if I do use the Client APIs I should be able to create an example that will both run as an app (calling `bokeh serve` first) or standalone (`python data_tables_server.py`). Is that correct?

I think I was trying to base my example off of `buttons_server.py` which is also in the `models` directory.

On a side note, I still can't get an app served using the Client APIs to work. I'm guessing it could be because I'm running my bokeh server on a separate \*nix box and then pulling the webpage from my regular computer. I think I'm setting everything up correctly, but nothing is rendering.. see below (Note: The `*.161` IP is that of the box running bokeh)

### Attempt at Client APIs
In one ssh terminal on the host I execute
```
BOKEH_DEV=true BOKEH_RESOURCES=server-dev bokeh serve --host localhost:5006 --host 192.168.200.161:5006
```
In another ssh terminal...
```
python3 buttons_server.py 
```

There is no output from the terminal that called `python3`. The output from the `bokeh serve` terminal is. 
```
INFO:bokeh.server.server:Starting Bokeh server version 0.12.1dev3-182-ga190697-dirty
DEBUG:bokeh.server.tornado:Allowed Host headers: ['localhost:5006', '192.168.200.161:5006']
DEBUG:bokeh.server.tornado:These host origins can connect to the websocket: ['localhost:5006', '192.168.200.161:5006']
DEBUG:bokeh.server.tornado:Patterns are:
DEBUG:bokeh.server.tornado:  [('/?',
DEBUG:bokeh.server.tornado:    <class 'bokeh.server.views.doc_handler.DocHandler'>,
DEBUG:bokeh.server.tornado:    {'application_context': <bokeh.server.application_context.ApplicationContext object at 0x7fb35d4a7160>,
DEBUG:bokeh.server.tornado:     'bokeh_websocket_path': '/ws'}),
DEBUG:bokeh.server.tornado:   ('/ws',
DEBUG:bokeh.server.tornado:    <class 'bokeh.server.views.ws.WSHandler'>,
DEBUG:bokeh.server.tornado:    {'application_context': <bokeh.server.application_context.ApplicationContext object at 0x7fb35d4a7160>,
DEBUG:bokeh.server.tornado:     'bokeh_websocket_path': '/ws'}),
DEBUG:bokeh.server.tornado:   ('/autoload.js',
DEBUG:bokeh.server.tornado:    <class 'bokeh.server.views.autoload_js_handler.AutoloadJsHandler'>,
DEBUG:bokeh.server.tornado:    {'application_context': <bokeh.server.application_context.ApplicationContext object at 0x7fb35d4a7160>,
DEBUG:bokeh.server.tornado:     'bokeh_websocket_path': '/ws'}),
DEBUG:bokeh.server.tornado:   ('/?',
DEBUG:bokeh.server.tornado:    <class 'bokeh.server.views.root_handler.RootHandler'>,
DEBUG:bokeh.server.tornado:    {'applications': {'/': <bokeh.server.application_context.ApplicationContext object at 0x7fb35d4a7160>},
DEBUG:bokeh.server.tornado:     'prefix': '',
DEBUG:bokeh.server.tornado:     'use_redirect': True}),
DEBUG:bokeh.server.tornado:   ('/static/(.*)',
DEBUG:bokeh.server.tornado:    <class 'bokeh.server.views.static_handler.StaticHandler'>)]
INFO:bokeh.command.subcommands.serve:Starting Bokeh server on port 5006 with applications at paths ['/']
INFO:bokeh.command.subcommands.serve:Starting Bokeh server with process id: 5742
INFO:bokeh.server.views.ws:WebSocket connection opened
DEBUG:bokeh.server.views.ws:Receiver created for Protocol('1.0')
DEBUG:bokeh.server.views.ws:ServerHandler created for Protocol('1.0')
INFO:bokeh.server.views.ws:ServerConnection created
DEBUG:bokeh.server.session:pushing doc to session '4oK0YzdUumQzzYLyEyBwKLziddpSE3K9Q7UBCPWywBcf'
INFO:tornado.access:200 GET / (192.168.200.76) 7.94ms
INFO:tornado.access:200 GET /static/css/bokeh.css?v=6d807b164c6043f4d0073237ed4185b0 (192.168.200.76) 3.17ms
INFO:tornado.access:200 GET /static/js/bokeh-compiler.js?v=3b2fe30f2fda4295433fe47c8929d006 (192.168.200.76) 6.49ms
INFO:tornado.access:200 GET /static/js/bokeh-widgets.js?v=67b2150bada9de057ae3c465d8f4316d (192.168.200.76) 9.81ms
INFO:tornado.access:200 GET /static/css/bokeh-widgets.css?v=82b53c925241b5847254f96e42783a01 (192.168.200.76) 15.41ms
INFO:tornado.access:200 GET /static/js/bokeh.js?v=25a8c367831633ba3ddb5cd02951243f (192.168.200.76) 43.55ms
INFO:tornado.access:200 GET /static/css/bokeh.css.map (192.168.200.76) 0.76ms
INFO:tornado.access:200 GET /static/css/bokeh-widgets.css.map (192.168.200.76) 19.28ms
INFO:tornado.access:200 GET /static/js/bokeh-compiler.js.map (192.168.200.76) 1.26ms
INFO:bokeh.server.views.ws:WebSocket connection opened
DEBUG:bokeh.server.views.ws:Receiver created for Protocol('1.0')
DEBUG:bokeh.server.views.ws:ServerHandler created for Protocol('1.0')
INFO:bokeh.server.views.ws:ServerConnection created
DEBUG:bokeh.server.session:Sending pull-doc-reply from session 'g3gU6O3g1xJITZ856TYz9jKWwW5US63uEbNW5L6DxgL9'
INFO:tornado.access:200 GET /static/js/bokeh-widgets.js.map (192.168.200.76) 44.74ms
INFO:tornado.access:200 GET /static/js/bokeh.js.map (192.168.200.76) 113.14ms
DEBUG:bokeh.server.tornado:[pid 5742] 2 clients connected
DEBUG:bokeh.server.tornado:[pid 5742]   / has 2 sessions with 0 unused
```

The JS Console is... (Nothing is rendered on the page, Running Chrome v52.0.2743.116 (64-bit) Mac OSX 10.11.6)
```
Bokeh: setting log level to: 'debug'
jsnlog.js:593Bokeh: Creating websocket 0 to 'ws://192.168.200.161:5006/ws' session 'g3gU6O3g1xJITZ856TYz9jKWwW5US63uEbNW5L6DxgL9'
jsnlog.js:578Bokeh: Websocket connection 0 is now open
jsnlog.js:593Bokeh: Pulling session for first time
jsnlog.js:593Bokeh: Creating Document from JSON
jsnlog.js:593Bokeh: Library versions: JS (0.12.2rc1)  /  Python (0.12.1dev3-182-ga190697-dirty)
jsnlog.js:593Bokeh: Created a new session from new pulled doc
embed.js:267Bokeh items were rendered successfully
```